src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1299,5 +1306,115 @@ public Task CompactDatabase()
+                Onprogress =>
+                {
+                    return Task.Run(async () =>
+                    {
+                        try
+                        {
+                            timer.Start();
+                            while (processDone.IsSet == false)
+                            {
+                                progress.Progress +=$"{await process.StandardOutput.ReadLineAsync()}{Environment.NewLine}";
+                                if (timer.ElapsedMilliseconds > 1000)
+                                {
+                                    Onprogress(progress);
+                                    progress.Progress = string.Empty;
+                                    timer.Restart();
+                                }
+                                ThrowIfTimeout(configuration.Timeout, timeout);
+
+                            }
+                            progress.Progress += $"{await process.StandardOutput.ReadToEndAsync()}{Environment.NewLine}";
+                            Onprogress(progress);
+                            if (process.ExitCode != 0)
+                            {
+                                throw new ApplicationException($"The data export tool have exited with code {process.ExitCode}.");
+                            }
+
+                            if (File.Exists(configuration.OutputFilePath) == false)
+                            {
+                                throw new FileNotFoundException($"Was expecting the output file to be located at {configuration.OutputFilePath}, but it is not there.");
+                            }
+
+                            using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                            using (var reader = File.OpenRead(configuration.OutputFilePath))
+                            using (var stream = new GZipStream(reader, CompressionMode.Decompress))
+                            {
+                                var source = new StreamSource(stream, context);
+                                var destination = new DatabaseDestination(database);                                
+                                var smuggler = new DatabaseSmuggler(database, source, destination, database.Time, result: result,onProgress: Onprogress);
+
+                                smuggler.Execute();
+                                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                                {
+                                    var json = result.ToJson();
+                                    context.Write(writer, json);
+                                }
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            result.AddError(e.ToString());
```

Comment:
Should we swallow stack here?

Reply:
how are we swallowing the stack?

Reply:
ok we don't. There is e.ToString(), not e.Message